src/main/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributor.java

```diff
@@ -144,16 +142,18 @@ private void distributeSampleUnits(CollectionExercise exercise, ExerciseSampleUn
         parent.setCollectionExerciseId(exercise.getId().toString());
         parent.setSampleUnitRef(sampleUnit.getSampleUnitRef());
         parent.setSampleUnitType(sampleUnit.getSampleUnitType().name());
-        parent.setPartyId(sampleUnit.getPartyId().toString());
+        parent.setId(sampleUnit.getSampleUnitId().toString());
+        parent.setPartyId(Objects.toString(sampleUnit.getPartyId(), null));
         parent.setCollectionInstrumentId(sampleUnit.getCollectionInstrumentId().toString());
         actionPlanId = collectionExerciseRepo
             .getActiveActionPlanId(exercise.getExercisePK(), sampleUnit.getSampleUnitType().name(),
                 exercise.getSurveyId());
       } else {
         SampleUnitChild child = new SampleUnitChild();
         child.setSampleUnitRef(sampleUnit.getSampleUnitRef());
+        child.setId(sampleUnit.getSampleUnitId().toString());
```

Comment:
Getting a NullPointerException around here when trying to execute collection exercises?